kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java

```diff
@@ -293,54 +274,70 @@ protected void deactivate(BundleContext context) {
+
+        quietUnregisterServlet(AUTH_RESOURCE_PATH);
+        quietUnregisterServlet(CONSOLE_RESOURCE_PATH);
+        quietUnregisterServlet(PASSWORD_AUTH_PATH);
+        quietUnregisterServlet(CERT_AUTH_PATH);
+        quietUnregisterServlet(LOGIN_MODULE_PATH + "/loginInfo");
+        quietUnregisterServlet(DENALI_MODULE_PATH + SESSION);
+        quietUnregisterServlet(LOGIN_MODULE_PATH + SESSION);
+        quietUnregisterServlet(LOGIN_MODULE_PATH + "/xsrf");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/xsrf");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/status");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/device");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/logservice");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/network");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/component");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/package");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/snapshot");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/certificate");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/security");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/users");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/file");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/device_snapshots");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/assetsUpDownload");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/log");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/skin");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/cloudservices");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/wires");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/wiresSnapshot");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/assetservices");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/extension");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/ssl");
+        quietUnregisterServlet(DENALI_MODULE_PATH + "/keystore");
+        quietUnregisterServlet(LOGIN_MODULE_PATH + "/extension");
+
+        if (this.wiresBlinkService != null) {
+            this.wiresBlinkService.stop();
+        }
+        quietUnregisterServlet(ADMIN_ROOT + "/sse");
+
+        if (this.eventService != null) {
+            this.eventService.stop();
+        }
+        quietUnregisterServlet(DENALI_MODULE_PATH + EVENT_PATH);
 
         for (final ServletRegistration reg : this.securedServlets) {
-            this.httpService.unregister(reg.path);
+            quietUnregisterServlet(reg.path);
         }
 
         for (final ServletRegistration reg : this.loginServlets) {
-            this.httpService.unregister(reg.path);
+            quietUnregisterServlet(reg.path);
+        }
+
+    }
+
+    private void quietUnregisterServlet(String path) {
+        try {
+            this.httpService.unregister(path);
+        } catch (final IllegalArgumentException iae) {
+            // nothing to do
```

Comment:
What is the reason for suppressing all IllegalArgumentExceptions?

Reply:
the logic has been unified and now the activation and the update call the same method, if the options are changed, a unregistration is made so if the aren't an exception is raised.

Reply:
If the problem is avoiding the unregistration when the component is activated, is it possible to check if the options are null and skip the unregistration step in that case?